### ISSUE TITLE
"switch" command no longer supported by Homebrew

### ISSUE THREAD

**Issue by JulienDutel** [rd2/tbd#58]
Refering to 
Read.me > Thermal Bridging & Derating (tbd) > TO DO > [MacOS (e.g. Catalina 10.15.4) Instructions](https://github.com/rd2/tbd#macos-eg-catalina-10154-instructions):

To instruct Homebrew to switch over to the right OpenSSL version, the switch command is [no longer supported](https://stackoverflow.com/questions/66082776/uncommon-homebrew-error-unknown-command-switch): 

`brew switch openssl 1.0.2t`

I haven't find the right syntax to replace this command line.

**Comment by rd2** [comment]
@JulienDutel : _Merci_ for raising this issue. I may have to reword the installation instructions ...

This in itself is not a recent change for Homebrew - it's been a year since they [dropped](https://github.com/rbenv/ruby-build/issues/1353#issuecomment-573414540) _switch_ (from what I understand). Yet the README instructions were written (and tested) less than a year ago on both an older and a more recent Macbook. And tested (successfully) on a new Macbook about 2 months ago. So I have trouble pinpointing why some have this issue (yet I don't). What version of Homebrew do you have (`brew -v`)? I have 2.7.4. and it still offers _switch_ (2-month old install). And your Command Line Tools (`clang -v`)? Some have resorted to switching to older versions of [clang](https://github.com/rbenv/ruby-build/issues/1353#issuecomment-710739814).

Are you installing on a new M1 machine? This does seem to be a real issue for [M1](https://github.com/rbenv/ruby-build/issues/1353#issuecomment-791727836).

Otherwise, we should be able to find a workaround.  Did you first manage to `brew install rbenv/tap/openssl@1.0`? _Tapping_ is a first crucial step. If so, have you also configured your environment as per the README instructions (namely the RUBY_CONFIGURE_OPTS)?

If you prefer, you can email me directly as we work this out. We can post the solution or workaround here afterwards. Up to you ...

**Comment by macumber** [comment]
Hi all, sorry I'm late.  I am not really a Mac user (primarily a Windows
user) so I don't know too much about brew.  Denis's questions all sounded
like the right ones to ask.  Sorry I don't have much else to offer, good
luck!

**Comment by JulienDutel** [comment]
@rd2 , _De rien_ et _merci également_. Here is quickly some answers to your questions:
 
>  What version of Homebrew do you have (`brew -v`)? I have 2.7.4. 

Homebrew 3.0.9 . I got this version by typing this command (recommended on the [Homebrew's website](https://brew.sh/index)) :
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
I guess this difference (3.0.9 vs 2.7.4) explain why I don't have the "switch" command.

> And your Command Line Tools (`clang -v`)? Some have resorted to switching to older versions of [clang]

Apple LLVM version 10.0.0 (clang-1000.10.44.4)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

> Are you installing on a new M1 machine? 

No, iMac with a good old fashion Intel i5.  Also, it run on  macOS HighSierra.

> Otherwise, we should be able to find a workaround. Did you first manage to `brew install rbenv/tap/openssl@1.0`? _Tapping_ is a first crucial step. 

It's been a week now since i've pass through it, but I'm pretty sure I did it.

> If so, have you also configured your environment as per the README instructions (namely the RUBY_CONFIGURE_OPTS)?

No, doesn't ring a bell.
